src/foam/core/types.js

```diff
@@ -404,6 +409,31 @@ foam.CLASS({
 
   properties: [
     [ 'factory', function() { return {} } ],
+    [
+      'comparePropertyValues',
+      function(o1, o2) {
+        if ( foam.typeOf(o1) != foam.typeOf(o2) ) return -1;
+
+        var keys1 = Object.keys(o1).sort();
```

Comment:
Sorry, I didn't write this part. For some reason, this got modified on my side. However, it is similar to part of the code already in types.js in foam2.